worker/uniter/runner/context/contextfactory.go

```diff
@@ -299,6 +299,12 @@ func (f *contextFactory) updateContext(ctx *HookContext) (err error) {
 	}
 	ctx.proxySettings = modelConfig.ProxySettings()
 
+	principal, _, err := ctx.unit.PrincipalUnit()
```

Comment:
we should really be using the "ok" variable, and only setting ctx.principal if it's true